src/osx/core/uilocale.mm

```diff
@@ -32,9 +32,24 @@
 #import <Foundation/NSLocale.h>
 #import <Foundation/NSDateFormatter.h>
 
+#include "wx/osx/private/uilocale.h"
+
 extern wxString
 wxGetInfoFromCFLocale(CFLocaleRef cfloc, wxLocaleInfo index, wxLocaleCategory cat);
 
+// Helper function to retrieve a pointer to the NSLocale of the current wxUILocale
+static NSLocale* gs_currentNSLocale = nullptr;
+
+NSLocale* wxGetCurrentNSLocale()
+{
+    static NSLocale* stdCLocale([[NSLocale localeWithLocaleIdentifier:@"en_US_POSIX"] retain]);
```

Comment:
This should be inside `if`, why even instantiate it if it's never going to be needed?

It's also going to leak currently, not sure if it's really important, but using `wxCFRef<NSLocale>` for it would avoid it.

Reply:
I moved the static locale into the if condition. And I used `wxCFRef`. However, I was not sure whether I have to keep the `retain` keyword or not.